src/OrchardCore.Modules/OrchardCore.Swagger/Startup.cs

```diff
@@ -0,0 +1,83 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Modules;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using System;
+using System.Collections.Generic;
+
+namespace OrchardCore.Swagger
+{
+    public class Startup : StartupBase
+    {
+        public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            app.UseSwagger();
+            app.UseSwaggerUI(c =>
+            {
+                var httpContextAccessor = serviceProvider.GetRequiredService<IHttpContextAccessor>();
+                var routeTenantPrefix = String.Empty;
```

Comment:
Not sure in which context this delgate can be used but if it is executed on each request, this `serviceProvider`, that is a scoped one when we build the pipeline, may not be the right one, e.g could be disposed.

So, i would have initialized once the `httpContextAccessor` variable outside and before `app.UseSwaggerUI()`, so that it references the `IHttpContextAccessor` tenant singleton instance. And then used it in the delegate. Note: Could be injected in the startup ctor.

Another solution here is to just use `ShellScope.Services`.

Reply:
Okay, sorry, i took a look at the source code, the delegate of `UseSwaggerUI()` is just a `setupAction` executed once, so no problem with your code using the passed `serviceProvider`.

Reply:
It's good idea to not be dependent on the specific implementation and hoist the variable, but I am going to leave it like this ;)

Reply:
Ok, did it anyway.